### PR TITLE
SONAR-8857 each organization can have a default qprofile for each lang

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java [hunk 1]
```diff
@@ -65,45 +67,47 @@
...
+  private final DefaultOrganizationProvider defaultOrganizationProvider;
 
   /**
    * To be kept when no ProfileDefinition are injected
    */
   public RegisterQualityProfiles(DbClient dbClient,
-    QProfileFactory profileFactory, RuleActivator ruleActivator, Languages languages, ActiveRuleIndexer activeRuleIndexer,
-    QProfileWsSupport qProfileWsSupport) {
-    this(dbClient, profileFactory, ruleActivator, Collections.emptyList(), languages, activeRuleIndexer, qProfileWsSupport);
+                                 QProfileFactory profileFactory, RuleActivator ruleActivator, Languages languages, ActiveRuleIndexer activeRuleIndexer,
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java]: bad indent

File: server/sonar-db-dao/src/test/java/org/sonar/db/qualityprofile/QualityProfileDaoTest.java [hunk 2]
```diff
@@ -140,11 +140,11 @@ public void find_all_is_sorted_by_profile_name() {
   public void get_default_profile() {
     dbTester.prepareDbUnit(getClass(), "shared.xml");
 
-    QualityProfileDto java = underTest.selectDefaultProfile(dbTester.getSession(), "java");
+    QualityProfileDto java = underTest.selectDefaultProfile(dbTester.getSession(), organization, "java");
```
Comment (simonbrandhof) [inline server/sonar-db-dao/src/test/java/org/sonar/db/qualityprofile/QualityProfileDaoTest.java]: the test should be improved to verify that condition on organization is applied. Currently the test passes if I drop `AND p.organization_uuid = #{organizationUuid, jdbcType=VARCHAR}` from SQL request.

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/SetDefaultActionTest.java [hunk 3]
```diff
@@ -137,8 +137,49 @@ public void set_default_profile_using_language_and_name() throws Exception {
...
-    checkDefaultProfile(xoo1Key, xoo1Profile.getKey());
-    checkDefaultProfile(xoo2Key, xoo2Profile.getKee());
+    checkDefaultProfile(organization, xoo1Key, xoo1Profile.getKey());
+    checkDefaultProfile(organization, xoo2Key, xoo2Profile.getKee());
+  }
+
+  @Test
+  public void should_not_change_other_organizations() throws Exception {
+    OrganizationDto organization1 = OrganizationTesting.newOrganizationDto();
+    db.organizations().insert(organization1);
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/SetDefaultActionTest.java]: Can be simply replaced by:
```
OrganizationDto organization1 = db.organizations().insert();
```

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/ExportAction.java [hunk 4]
```diff
@@ -127,8 +130,9 @@ private void writeResponse(QualityProfileDto profile, @Nullable String exporterK
 
   private QualityProfileDto getProfile(@Nullable String name, String language) {
     try (DbSession dbSession = dbClient.openSession(false)) {
-      QualityProfileDto profile = name == null ? dbClient.qualityProfileDao().selectDefaultProfile(dbSession, language)
-        : dbClient.qualityProfileDao().selectByNameAndLanguage(name, language, dbSession);
+      OrganizationDto organization = qProfileWsSupport.getDefaultOrganization(dbSession);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/ExportAction.java]: MINOR Remove this use of "getDefaultOrganization"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java [hunk 5]
```diff
@@ -65,45 +67,47 @@
...
+                                 QProfileWsSupport qProfileWsSupport, DefaultOrganizationProvider defaultOrganizationProvider) {
+    this(dbClient, profileFactory, ruleActivator, Collections.emptyList(), languages, activeRuleIndexer, qProfileWsSupport, defaultOrganizationProvider);
   }
 
   public RegisterQualityProfiles(DbClient dbClient,
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java]: MAJOR Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java [hunk 6]
```diff
@@ -65,45 +67,47 @@
...
     QProfileFactory profileFactory, RuleActivator ruleActivator,
-    List<ProfileDefinition> definitions, Languages languages, ActiveRuleIndexer activeRuleIndexer, QProfileWsSupport qProfileWsSupport) {
+    List<ProfileDefinition> definitions, Languages languages, ActiveRuleIndexer activeRuleIndexer, QProfileWsSupport qProfileWsSupport, DefaultOrganizationProvider defaultOrganizationProvider) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java]: MINOR Split this 194 characters long line (which is greater than 180 authorized). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java [hunk 7]
```diff
@@ -65,45 +67,47 @@
   private final Languages languages;
   private final ActiveRuleIndexer activeRuleIndexer;
   private final QProfileWsSupport qProfileWsSupport;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java]: MAJOR Remove this unused "qProfileWsSupport" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SetDefaultAction.java [hunk 8]
```diff
@@ -65,15 +67,17 @@ public void handle(Request request, Response response) {
     QProfileReference reference = QProfileReference.from(request);
     try (DbSession dbSession = dbClient.openSession(false)) {
       QualityProfileDto qualityProfile = qProfileWsSupport.getProfile(dbSession, reference);
+      OrganizationDto organization = dbClient.organizationDao().selectByUuid(dbSession, qualityProfile.getOrganizationUuid())
+        .orElseThrow(() -> new IllegalStateException(format("Cannot find organization '%s' for quality profile '%s'", qualityProfile.getOrganizationUuid(), qualityProfile.getKee())));
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SetDefaultAction.java]: MINOR Split this 183 characters long line (which is greater than 180 authorized). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [QualityProfileDao.java#L158](https://github.com/SonarSource/sonarqube/blob/5e4e2035a4c139ad59e1a9e16ab72dd7546035ec/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L158): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L171](https://github.com/SonarSource/sonarqube/blob/5e4e2035a4c139ad59e1a9e16ab72dd7546035ec/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L171): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/SetDefaultActionTest.java (hunk 3 above)
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/SetDefaultActionTest.java]: Done.

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java (hunk 1 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java]: Fixed.

File: server/sonar-db-dao/src/test/java/org/sonar/db/qualityprofile/QualityProfileDaoTest.java (hunk 2 above)
Comment (bartfastiel) [inline server/sonar-db-dao/src/test/java/org/sonar/db/qualityprofile/QualityProfileDaoTest.java]: Done.

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java (hunk 7 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java]: Done.

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java (hunk 6 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java]: Done.

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SetDefaultAction.java (hunk 8 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SetDefaultAction.java]: Done.

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java [hunk 9]
```diff
@@ -64,46 +63,45 @@
...
 
-  public RegisterQualityProfiles(DbClient dbClient,
-    QProfileFactory profileFactory, RuleActivator ruleActivator,
-    List<ProfileDefinition> definitions, Languages languages, ActiveRuleIndexer activeRuleIndexer, QProfileWsSupport qProfileWsSupport) {
+  public RegisterQualityProfiles(DbClient dbClient, QProfileFactory profileFactory, RuleActivator ruleActivator,
+    List<ProfileDefinition> definitions,
+    Languages languages, ActiveRuleIndexer activeRuleIndexer, DefaultOrganizationProvider defaultOrganizationProvider) {
     this.dbClient = dbClient;
     this.profileFactory = profileFactory;
     this.ruleActivator = ruleActivator;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java]: MAJOR Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java [hunk 10]
```diff
@@ -64,46 +63,45 @@
...
     this.definitions = definitions;
     this.languages = languages;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java]: MINOR Split this 194 characters long line (which is greater than 180 authorized). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java [hunk 11]
```diff
@@ -64,46 +63,45 @@
   private final RuleActivator ruleActivator;
   private final Languages languages;
   private final ActiveRuleIndexer activeRuleIndexer;
-  private final QProfileWsSupport qProfileWsSupport;
+  private final DefaultOrganizationProvider defaultOrganizationProvider;
 
   /**
    * To be kept when no ProfileDefinition are injected
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java]: MAJOR Remove this unused "qProfileWsSupport" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SetDefaultAction.java [hunk 12]
```diff
@@ -65,15 +67,18 @@ public void handle(Request request, Response response) {
     QProfileReference reference = QProfileReference.from(request);
     try (DbSession dbSession = dbClient.openSession(false)) {
       QualityProfileDto qualityProfile = qProfileWsSupport.getProfile(dbSession, reference);
+      OrganizationDto organization = dbClient.organizationDao().selectByUuid(dbSession, qualityProfile.getOrganizationUuid())
+        .orElseThrow(() -> new IllegalStateException(
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SetDefaultAction.java]: MINOR Split this 183 characters long line (which is greater than 180 authorized). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [QualityProfileDao.java#L158](https://github.com/SonarSource/sonarqube/blob/b0cd512e0d316b0bde629ca479b369692ced45d6/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L158): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L171](https://github.com/SonarSource/sonarqube/blob/b0cd512e0d316b0bde629ca479b369692ced45d6/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L171): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [QualityProfileDao.java#L158](https://github.com/SonarSource/sonarqube/blob/b0cd512e0d316b0bde629ca479b369692ced45d6/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L158): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityProfileDao.java#L171](https://github.com/SonarSource/sonarqube/blob/b0cd512e0d316b0bde629ca479b369692ced45d6/server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java#L171): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request merged**